tensorflow/python/keras/callbacks.py

```diff
@@ -2068,6 +2069,12 @@ def _stop_trace(self, batch=None):
     profiler.stop()
     self._is_tracing = False
 
+  def _collect_learning_rate(self, logs):
+    lr_schedule = getattr(self.model.optimizer, "lr", None)
```

Comment:
Could you retrieve `_decayed_lr` instead? Then the `isinstance` call etc. is taken care of, as well as any potential decay.

Reply:
I didn't use `_decayed_lr` here initially since it is a private API that might not behave as expected in all cases.
E.g. using `_decayed_lr` would break usage with `tf.keras.mixed_precision.experimental.LossScaleOptimizer` since it  only wraps an optimizer and would fail when trying to access `self._hyper["learning_rate"]`. Also if users opt to implement optimizers that don't rely on a `learning_rate` hyperparameter to apply gradient updates, `_decayed_lr` would fail as well. Using `getattr(self.model.optimizer, "lr", None)` covers both cases, so I would prefer to keep it as, or am I missing something?

Reply:
Thanks for the clarification. I don't think you're missing anything. To be clear: you choose to retrieve the learning rate schedule (or constant, or callable) rather than the value that would actually be used during training (the value from `model._decayed_lr` that might have a decay applied)?

Reply:
Yes, I wasn't aware of the `decay` keyword argument, but it looks like it is only included for backward compatibility.